core/src/main/java/hudson/tasks/LogRotator.java

```diff
@@ -297,6 +299,16 @@ private String toString(Integer i) {
         if (i==null || i==-1)   return "";
         return String.valueOf(i);
     }
+    
+    @Override
+    public String toString() {
+    	return Objects.toStringHelper( this )
```

Comment:
Added a human-readable ToString to LogRotator. When a globally-defined LogRotator is applied to a project (conceivably resulting in the _deletion_ of build records), we emit a very complete message to the Jenkins log recorder, which includes the LogRotator used. This toString makes it easier for that printout to show exactly which rules were applied.